lib/auth/init.go

```diff
@@ -281,6 +278,36 @@ func Init(cfg InitConfig, dynamicConfig bool) (*AuthServer, *Identity, error) {
 	return asrv, identity, nil
 }
 
+// setClusterName will set the cluster name resource if it doesn't exist on
+// the backend. if it does exist on the backend and differs from what is in
+// the configuration file we will hardfail. if it matches, we do nothing.
+func setClusterName(cfg InitConfig, asrv *AuthServer) error {
+	cn, err := asrv.GetClusterName()
+	if err != nil && !trace.IsNotFound(err) {
+		return trace.Wrap(err)
+	}
+
+	// if nothing is set on the backend, set and return
+	if trace.IsNotFound(err) {
+		err = asrv.SetClusterName(cfg.ClusterName)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		log.Infof("[INIT] Updating Cluster Configuration: %v", cfg.ClusterName)
+
+		return nil
+	}
+
+	// we found a cluster name set on the backend, but it differs from is in
+	// the config file: hard fail.
+	if cn.GetClusterName() != cfg.ClusterName.GetClusterName() {
+		return trace.BadParameter("unable to update cluster configuration, cluster name already set: %v", cn)
```

Comment:
I would make this call to action: `Cluster name in configuration (%v) differs from cluster configuration(%v). Update your local server configuration cluster name to %v and restart the server.`

Reply:
@klizhentas we're going with _cannot rename cluster "turing" to "pascal": clusters cannot be renamed_

Reply:
@kontsevoy I would still suggest to add some call to action here, e.g. update your local confiugration to something, otherwise users will ask us what to do :)

Reply:
The only way to see this error is when you change the config file and restart. I just have gone through the process myself. Saying "cannot rename A to B" makes it crystal clear _which_ settings have changed. "local configuration" is confusing. I _know_ what I touched. I just need to know which setting (among many, because I just enabled auth section too!) has triggered the error.  "local configuration" is not an explained or documented concept.

Reply:
the call to action is to undo what I just did. but not all of it, just one value.

Reply:
up to you